### ISSUE TITLE
i want to use grayscale image and mask image.. i want big help from you ㅠㅡㅠ

### ISSUE THREAD

**Issue by skyoung2811** [milesial/Pytorch-UNet#194]
![Screenshot from 2020-06-30 10-32-30](https://user-images.githubusercontent.com/34771635/86072668-0a24f180-babd-11ea-846c-c7ed5b9d9143.png)

i want to use image like grayscale input and mask input.. but there is dimention error how can i use grayscale image?

**Comment by milesial** [comment]
If you have only one input channel, then you have to change to parameters of the construction of the Unet.
https://github.com/milesial/Pytorch-UNet/blob/1c6a9505c964fdd1bc7d55887725551afc94b956/train.py#L157
change n_channels to 1

**Comment by skyoung2811** [comment]
![Screenshot from 2020-06-30 11-45-39](https://user-images.githubusercontent.com/34771635/86077630-452c2280-bac7-11ea-8a21-590ca0995abc.png)

I changed but it has  problem..

**Comment by milesial** [comment]
Make sure you are using greyscale images and loading them as such in https://github.com/milesial/Pytorch-UNet/blob/1c6a9505c964fdd1bc7d55887725551afc94b956/utils/dataset.py#L54

**Comment by skyoung2811** [comment]
that is same.. first i retry first time. the showing is below..
![Screenshot from 2020-06-30 13-22-54](https://user-images.githubusercontent.com/34771635/86083331-e9689600-bad4-11ea-81a0-dd2bfa60054b.png)

it's my image's problem?

**Comment by milesial** [comment]
Change your input channels to 1!
For the loss, the target is the true masks and the input is the predicted masks.
Here your target size is wrong, meaning you're loading your masks wrong.

**Comment by skyoung2811** [comment]
Now, i exchaned all mask image.. but it has same problem.. ㅠㅡㅠ

**Comment by milesial** [comment]
PNG or any format is fine as long as you transform the loaded masks and images to have to needed dimensionality.
Look at the Pillow documentation to make sure you loaded your masks correctly.

**Comment by skyoung2811** [comment]
I solved it. but after trainning, how to direct trainning  one picture? 
i want to direct make a effort about other picture. thanks.

**Comment by milesial** [comment]
If you want to predict with predict.py, read the README, it's explained there.

**Comment by skyoung2811** [comment]
After training your model and saving it to MODEL.pth

how to make MODEL.pth?

**Comment by milesial** [comment]
By training the model with train.py. Please read the full README before opening issues.
